Lib/test/test_exceptions.py

```diff
@@ -1170,6 +1170,12 @@ def inner():
             self.fail("RecursionError not raised")
         self.assertEqual(wr(), None)
 
+    def test_errno_ENOTEMPTY(self):
+        with self.assertRaises(OSError) as cm:
+            test_dir = os.path.dirname(__file__)
+            os.rmdir(test_dir)
```

Comment:
I would prefer to create a temporary directory rather than trying to remove os.path.dirname(__file__) which can be /usr/lib64/python3.7/test/ for example! You may get a permission error.